madanalysis/layout/plotflow_for_dataset.py

```diff
@@ -84,58 +88,133 @@ def CreateHistogram(self):
-            if select.stack == StackingMethodType.NORMALIZE2ONE or (
+            if self.main.selection[
+                iabshisto
+            ].stack == StackingMethodType.NORMALIZE2ONE or (
                 self.main.stack == StackingMethodType.NORMALIZE2ONE
                 and self.main.selection[iabshisto].stack == StackingMethodType.AUTO
             ):
+                integral = (
+                    self.histos[iplot].positive.integral
+                    - self.histos[iplot].negative.integral
+                )
+                multiweight_integral = 0
+                if self.multiweight_histos[iplot]:
+                    multiweight_integral = self.multiweight_histos[iplot].integral
                 if integral > 0.0:
                     scale = 1.0 / integral
                 else:
                     scale = 0.0
+                if multiweight_integral > 0.0:
+                    multiweight_scale = 1.0 / multiweight_integral
 
             # Case 2: No normalization
             elif self.main.normalize == NormalizeType.NONE:
                 scale = 1.0
+                multiweight_scale = 1.0
 
             # Case 3 and 4 : Normalization formula depends on LUMI
             #                or depends on WEIGHT+LUMI
             elif self.main.normalize in [NormalizeType.LUMI, NormalizeType.LUMI_WEIGHT]:
 
+                # integral
+                integral = (
+                    self.histos[iplot].positive.integral
+                    - self.histos[iplot].negative.integral
+                )
+                multiweight_integral, multiweight_eff = 0, 0
+                if len(self.multiweight_histos) != 0:
+                    multiweight_integral = self.multiweight_histos[iplot].integral
+
+                # compute efficiency : Nevent / Ntotal
+                if self.dataset.measured_global.nevents == 0:
+                    eff = 0
+                else:
+                    nevt = float(self.dataset.measured_global.nevents)
+                    eff = (
+                        self.histos[iplot].positive.nevents
+                        + self.histos[iplot].negative.nevents
+                    ) / nevt
+                    if self.multiweight_histos[iplot]:
+                        multiweight_eff = self.multiweight_histos[iplot].nevents / nevt
+
                 # compute the good xsection value
                 thexsection = self.xsection
                 if self.main.normalize == NormalizeType.LUMI_WEIGHT:
                     thexsection = thexsection * self.dataset.weight
 
-                scale = processor.scale(lumi=self.main.lumi)
+                # compute final entries/event ratio
+                entries_per_events = 0
+                sumw = self.histos[iplot].positive.sumw - self.histos[iplot].negative.sumw
+                Nentries = (
+                    self.histos[iplot].positive.sumwentries
+                    - self.histos[iplot].negative.sumwentries
```

Comment:
Not a + sign?

Reply:
That is not from me, but negative `sumwentries` are absolute valued, as far as I know, so don't we need to subtract them?

Reply:
This quantity is needed to calculate the mean number of entries by event. Then, we need to sum the two absolute values, if I am correct.